src/o365/graph/commands/teams/teams-clone.ts

```diff
@@ -0,0 +1,187 @@
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  teamId: string;
+  displayName: string;
+  mailNickname: string;
+  partsToClone: string;
+  description?: string;
+  classification?: string;
+  visibility?: string;
+}
+
+class GraphTeamsCloneCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.TEAMS_CLONE}`;
+  }
+
+  public get description(): string {
+    return 'Creates a copy of a Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.description = typeof args.options.description !== 'undefined';
+    telemetryProps.classification = typeof args.options.classification !== 'undefined';
+    telemetryProps.visibility = typeof args.options.visibility !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): request.RequestPromise => {
+
+        const requestOptions: any = {
+          url: `${auth.service.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/clone`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            "content-type": "application/zip",
+            accept: 'application/json;odata.metadata=none'
+          }),
+          json: true,
+          body: {
+            displayName : args.options.displayName,
+            mailNickname : args.options.mailNickname,
+            partsToClone : args.options.partsToClone,
+            description : args.options.description || undefined,
+            classification : args.options.classification || undefined,
+            visibility : args.options.visibility || undefined,
+          }
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        return request.post(requestOptions);
+      })
+      .then((res: any): void => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+
+        cb();
+      }, (res: any): void => this.handleRejectedODataJsonPromise(res, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --teamId <teamId>',
+        description: 'The ID of the Microsoft Teams team to clone'
+      },
+      {
+        option: '-n, --displayName <displayName>',
+        description: 'The display name for the new Microsoft Teams Team'
+      },
+      {
+        option: '-m, --mailNickname <mailNickname>',
```

Comment:
Let us wait for Waldek's response

Reply:
@waldekmastykarz , ping.

Reply:
If the property doesn't work, what's the purpose of including in the CLI? To avoid confusion, we could remove it for now and introduce it back once it's working in the API.

Reply:
@waldekmastykarz , my initial thought was the same, however it is required in the payload and the graph complains if it is not present. :)

Reply:
We can still include it in the body and don't expose it as an option in the CLI :)

Reply:
Sweet.

Reply:
@VelinGeorgiev, I have included a note in the description. Please review.

Reply:
@sprider , I actually think the suggestion from @waldekmastykarz  is better. We can remove it from the options and assign a default value until they fix it. For example if the user specifies for displayName "My cool team", we can escape the special characters and trim the whitespace so the nickname can look like "Mycoolteam" then send that as mailNickname with the payload. What do you think?

Reply:
> @sprider , I actually think the suggestion from @waldekmastykarz is better. We can remove it from the options and assign a default value until they fix it. For example if the user specifies for displayName "My cool team", we can escape the special characters and trim the whitespace so the nickname can look like "Mycoolteam" then send that as mailNickname with the payload. I would escape everything that is not number or char (a-zA-Z0-9). What do you think?